src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,131 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int START_CAPACITY = 10;
+    private static final int EMPTY = 0;
+    private static final double GROW = 1.5;
+    private int size;
+    private int capacity;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[START_CAPACITY];
+        size = EMPTY;
+        capacity = START_CAPACITY;
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            grow();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("IndexOutOfBounds");
+        } else {
+            if (size == capacity) {
+                grow();
+            }
+            T[] newArray = (T[]) new Object[capacity];
+            System.arraycopy(array, 0, newArray,0, index);
+            newArray[index] = value;
+            System.arraycopy(array, index, newArray,index + 1, size - index);
+            array = newArray;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        if (size + list.size() > capacity) {
+            grow();
+        }
+        T[] acceptedArray = (T[]) new Object[list.size()];
+        for (int i = 0; i < acceptedArray.length; i++) {
+            acceptedArray[i] = list.get(i);
+        }
+        System.arraycopy(acceptedArray, 0, array, size, acceptedArray.length);
+        size += acceptedArray.length;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (legalBound(index)) {
+            throw new ArrayListIndexOutOfBoundsException("IndexOutOfBounds");
+        } else {
+            return array[index];
+        }
```

Comment:
Please move exception throwing inside method legalBound(), give this method proper name and code will be much more clear
```suggestion
        checkIndex(index)
        return array[index];
```
Please recheck other methods with same situation